recurring_contract/models/contract_group.py

```diff
@@ -304,57 +311,87 @@ def _calculate_start_date_and_offset(self):
+                (
+                    "line_ids.product_id",
+                    "in",
+                    contract.product_ids.ids,
+                )
+            ]
+        else:
+            search_filter = [
                 "&",
                 "&",
                 "&",
                 "&",
-                ("move_type", "=", "out_invoice"),
-                ("state", "=", "cancel"),
+                "&",
+                "|",
                 ("payment_state", "not in", ["paid", "not_paid"]),
+                ("state", "=", "cancel"),
+                ("invoice_date_due", "=", invoicing_date),
+                ("partner_id", "=", self.partner_id.id),
+                ("move_type", "=", "out_invoice"),
                 ("line_ids.contract_id", "in", self.active_contract_ids.ids),
                 (
                     "line_ids.product_id",
                     "in",
                     self.active_contract_ids.mapped("product_ids").ids,
-                ),
-            ]
-        )
-        # Check for contract group suspension
-        is_suspended = (
-            self.invoice_suspended_until
-            and self.invoice_suspended_until > invoicing_date
-        )
+                )]
+
+        existing_invoices = self.env["account.move"].search(search_filter)
 
-        return bool(dangling_invoices) or is_suspended
+        is_sub_proposal = contract is not None and contract.source_id is 554
+
+        # Check for contract group suspension when no specific contract is given from a sub proposal is given
+        if is_sub_proposal:
+            return bool(existing_invoices)
+        else:
+            is_suspended = (
+                self.invoice_suspended_until
+                and self.invoice_suspended_until > invoicing_date
+            )
+            return bool(existing_invoices) or is_suspended
 
-    def _process_invoice_generation(self, invoicer, invoicing_date):
+    def _process_invoice_generation(self, invoicer, invoicing_date, contract=None):
         self.ensure_one()
         active_contracts = self.active_contract_ids
-        open_invoices = active_contracts.mapped("open_invoice_ids").filtered(
-            lambda i: i.invoice_date_due >= invoicing_date
+        open_invoices = self.active_contract_ids.mapped("open_invoice_ids").filtered(
+            lambda i: i.invoice_date_due >= invoicing_date and i.invoice_date_due.year == invoicing_date.year
         )
 
         # invoice already open we complete the move lines
         current_rec_unit_date = getattr(invoicing_date, self.recurring_unit)
         # Keep invoice from the same month or year
         # (depending on the recurring unit)
         open_invoice = open_invoices.filtered(
-            lambda m: getattr(m.invoice_date_due, self.recurring_unit)
-            == current_rec_unit_date
+            lambda m: getattr(m.invoice_date_due, self.recurring_unit) == current_rec_unit_date
         )
+        if len(open_invoice) > 1:
```

Comment:
@ecino I'm not completely sure if we need this check but it seems weird to have multiple open invoices for the same invoice date. I had this locally, I don't know if it is the case in production.

Plus if there are multiple open invoices, if will add to those invoices a new line for this contract, so the sponsor would pay multiple times for the same contract.

Reply:
We decided to raise en error so that we can handle that case whe it comes up.